packages/performance/src/services/remote_config_service.ts

```diff
@@ -181,11 +194,47 @@ function processConfig(
   } else if (SECONDARY_CONFIGS.logSource) {
     settingsServiceInstance.logSource = SECONDARY_CONFIGS.logSource;
   }
+
   if (entries.fpr_log_endpoint_url) {
     settingsServiceInstance.logEndPointUrl = entries.fpr_log_endpoint_url;
   } else if (SECONDARY_CONFIGS.logEndPointUrl) {
     settingsServiceInstance.logEndPointUrl = SECONDARY_CONFIGS.logEndPointUrl;
   }
+
+  // Key from Remote Config has to be non-empty string, otherwsie use local value.
+  if (entries.fpr_log_transport_key) {
+    settingsServiceInstance.transportKey = entries.fpr_log_transport_key;
+  } else if (SECONDARY_CONFIGS.transportKey) {
+    settingsServiceInstance.transportKey = SECONDARY_CONFIGS.transportKey;
+  }
+
+  // If config object state indicates that no template has been set, that means it is new user of
+  // Performance Monitoring and should use the old log endpoint, because it is guaranteed to work.
+  if (
+    state === undefined ||
+    state === 'INSTANCE_STATE_UNSPECIFIED' ||
+    state === 'NO_TEMPLATE'
+  ) {
+    if (NO_TEMPLATE_CONFIGS.shouldSendToTransport !== undefined) {
```

Comment:
`NO_TEMPLATE_CONFIGS.shouldSendToTransport` can be false as boolean value. If changing the check above into `if (NO_TEMPLATE_CONFIGS.shouldSendToTransport)`, then this check will return false if `shouldSendToTransport` is false. It will cause boolean value assignment inside the `if` block being skipped. Please correct me if I was wrong understanding the comment. Thank you!